plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java

```diff
@@ -307,6 +303,17 @@ public static void checkLibraryList(IJavaProject javaProject, IProgressMonitor m
     }
   }
 
+  /** Find the master container in the given project or {@code null} if none. */
+  public static IClasspathEntry findMasterContainer(IJavaProject javaProject)
+      throws JavaModelException {
+    IClasspathEntry[] rawClasspath = javaProject.getRawClasspath();
+    return Stream.of(rawClasspath)
+        .filter(LibraryClasspathContainer::isEntry)
```

Comment:
`::isEntry` is basically redundant with below, right?

Reply:
Almost — `isEntry()` also tests the containerKind.